src/e2e/puppeteer/__tests__/gesture-alert.ts

```diff
@@ -0,0 +1,139 @@
+import path from 'path'
+import { KnownDevices } from 'puppeteer'
+import sleep from '../../../util/sleep'
+import configureSnapshots from '../configureSnapshots'
+import screenshot from '../helpers/screenshot'
+import waitForFrames from '../helpers/waitForFrames'
+import { page } from '../setup'
+
+expect.extend({
+  toMatchImageSnapshot: configureSnapshots({ fileName: path.basename(__filename).replace('.ts', '') }),
+})
+
+vi.setConfig({ testTimeout: 20000, hookTimeout: 20000 })
+
+/**
+ * Helper function to smoothly move touch from one point to another.
+ * Creates a series of touchMove events to simulate realistic gesture movement.
+ *
+ * @param x1 - Starting X coordinate.
+ * @param y1 - Starting Y coordinate.
+ * @param x2 - Ending X coordinate.
+ * @param y2 - Ending Y coordinate.
+ */
+const move = async (x1: number, y1: number, x2: number, y2: number): Promise<void> => {
```

Comment:
This is a great idea, but you can use the existing [swipe](https://github.com/cybersemics/em/blob/9889f4a15288d82c787e2ea5b4e3ef976f8fe4f3/src/e2e/puppeteer/helpers/swipe.ts) helper. You'll just need to extend it to work in multiple directions.